source/core/Events.cpp

```diff
@@ -376,21 +390,71 @@ class DynamicEventRegInfo {
         _entry.reserve(nEntries);
     }
     size_t NumEntries() const { return _entry.size(); }
-    Int32 DynamicEventMatch(const EventCommonData &eData) {
+
+    // Recursive function to match a registered dynamic event with polymorphic data (clusters/arrays of refs)
+    // If the registered data is a cluster or nested cluster updated *dynIndexBase with the nested element
+    // index as it traverses, and returns a non-zero index if a match occcurs.  (The dynIndex value is
+    // the same one storied in the EventSpec in the event structure configuration data.)
+    Int32 DynamicEventMatchCore(TypeRef regRefType, void *pData, RefNum refnum, Int32 *dynIndexBase) {
         Int32 dynIndex = 0;
-        for (size_t i = 0; i < _entry.size(); i++) {
+        if (regRefType->IsRefnum() && pData && ((RefNumVal*)pData)->GetRefNum() == refnum) {
+            dynIndex = *dynIndexBase;
+            return dynIndex;
+        } else if (regRefType->IsArray() && regRefType->Rank() == 1 && regRefType->GetSubElement(0)->IsRefnum()) {
```

Comment:
Do you know if the VI is broken in NG when the user attempts to connect an N-Dimensional array?